pkg/cmd/gen_driftignore_test.go

```diff
@@ -55,53 +54,37 @@ func TestGenDriftIgnoreCmd_Input(t *testing.T) {
 			output: "./testdata/output_stdin_valid_filter2.txt",
 			err:    errors.New("open doesnotexist: no such file or directory"),
 		},
-		{
-			name:   "test error when input flag is not specified",
-			args:   []string{},
-			output: "",
-			err:    errors.New("Error: you must specify an input to parse JSON from. Use driftctl gen-driftignore -i <drifts.json>\nGenerate a JSON file using the output flag: driftctl scan -o json://path/to/drifts.json"),
-		},
```

Comment:
This test cannot be simply dropped, we should now ensure that it will not trigger any error if we do not pass any args.

Reply:
Sorry, I read too quickly, the note I linked isn't as related to this thread as I thought.

If we want to test the default-stdin case, we'll have to do some ceremony around replacing stdin during the test. I don't think that would assist design much, or do much to prevent regressions: it seems like all it would really do is ensure golang stdlib's `os.Stdin` does what it says it does.

Rephrased in code comments:

```go

// This line is covered by tests, but this value is never used.
// This compiles, because `os.Stdin` is a `*os.File`.
// We do not test that os.Stdin behaves like a file, but surely not every program that
// uses go's stdlib needs to re-validate it completely?
driftFile := os.Stdin

// The contents of this branch are covered by tests
if opts.InputPath != "-" {
	var err error
	driftFile, err = os.Open(opts.InputPath)
	if err != nil {
		return err
	}
	defer driftFile.Close()
}
```

This PR already removes similar test setup around pipes and stdout, instead using temporary files passed to `-o` to avoid the need for it.

Reply:
I think we should keep this test since this will allow us to assert on the results by the same time. Maybe we can drop this case  `{args: []string{"gen-driftignore", "--input", "-"}}` in the other test and keep this one. The other test is only validating that we are not getting any error, but obviously `TestGenDriftIgnoreCmd_Input` supersed it.

We should have to update a little bit this test to be able to send a proper input to stdin

Reply:
For sure we do not care about validating `os.Stdin` behavior here.
The point is more about to validate that when we do not specify anything we try to read from `os.Stdin`, replacing it during the test will cover that.

The problem I have here is that if we change the default `driftFile := os.Stdin` we are blind in test suite about this, and since it could be a potential big BC break I just wanted to cover that.

Maybe you have a better approach for that ?

Reply:
That line is arguably more like config than code. It contains no logic. If we were to test it by swapping out `os.Stdin`, the test isn't really proving much: we're basically just asserting that the symbols on each side of the test-prod boundary match, if you see what I mean.

I'm not saying there's no value in such a test, and am also not saying that _not_ covering the stdin case is risk-free: what I am saying is that subjectively, I suspect that such a test is on the wrong side of the cost/benefit calculus, _given the way the test harness is already set up_. I wouldn't be making this point if we were talking about a more "integration test" style harness, against the compiled binary, that we could pipe _real_ stdin into (not just stubbing out `os.Stdin` in go). As far as I know driftctl doesn't have such a harness?

I don't want to be too stubborn though, this is my first contribution to this project! I've made my case, let me know what you decide.

Reply:
> we're basically just asserting that the symbols on each side of the test-prod boundary match, if you see what I mean.

Yeah totally.

I agree with your point about the value in a such test.
Maybe here we do not want an unit test but something more close to an acceptance test or E2E testing.

> I wouldn't be making this point if we were talking about a more "integration test" style harness, against the compiled binary, that we could pipe real stdin into (not just stubbing out os.Stdin in go). As far as I know driftctl doesn't have such a harness?

Ok ok my bad I read that comment too fast and that was exactly my feelings too in my previous sentence 😅  Actually we do have some integration testing (we call that acceptance testing but whatever). We basically run driftctl cmd in a wrapper that is able to expose an assert API, you can have an example [here](https://github.com/cloudskiff/driftctl/blob/main/pkg/resource/google/google_bigtable_table_test.go#L11)

I want to cover this default behavior case from the end user point of view, so you are right this does not make sense to cover this in a kinda unit scoped test 👍🏻 

Actually we do not have any mechanism to be able to replace the `os.Stdin` in out "micro-acceptance-test-framework" but this is something we can probably do without any trouble. We are capturing driftctl output during the run to be able to assert on it (btw we do not have any acceptance test case that is asserting on stdout as far as I know 😅 ). 

Soo maybe this is something we can achieve in another PR, but I don't think that the fix will be too heavy so as you like. Since it will do not put main branch in an inconsistent state if we split this in another PR, I'd rather prefer that. I'm I'm not teaching you anything but big PR can be really painfull to review